conans/client/cmd/export.py

```diff
@@ -236,12 +236,18 @@ def _detect_scm_revision(path):
         return None, None
 
     repo_obj = SCM.availables.get(repo_type)(path)
-    return repo_obj.get_revision(), repo_type
+    try:
+        revision = repo_obj.get_revision()
+        return revision, repo_type
+    except Exception as exc:
+        if revisions_enabled:
+            output.info("Cannot detect revision from SCM: {}".format(exc))
+        return None, None
 
 
 def _update_revision_in_metadata(package_layout, revisions_enabled, output, path, digest):
 
-    scm_revision_detected, repo_type = _detect_scm_revision(path)
+    scm_revision_detected, repo_type = _detect_scm_revision(path, revisions_enabled, output)
```

Comment:
I would move the try/catch here. Although it is _private_, I think that the function `_detect_scm_revision` should raise if something goes wrong for the `scm`, so here it will be more explicit the fallback to `hash` behavior: